src/bot.py

```diff
@@ -273,10 +273,14 @@ def on_maintenance(self):
 
         # Check if we are out of tps or need repairing
         need_repair = self._ui_manager.repair_needed()
-        if self._tps_left < random.randint(3, 5) or need_repair or self._config.char["always_repair"]:
+        if self._tps_left < random.randint(3, 5) or need_repair or self._config.char["always_repair"] or self._config.char["always_refill_keys"] & self._config.routes["run_arcane"]:
             if need_repair: Logger.info("Repair needed. Gear is about to break")
             else: Logger.info("Repairing and buying TPs at next Vendor")
-            self._curr_loc = self._town_manager.repair_and_fill_tps(self._curr_loc)
+            if self._config.char["always_refill_keys"] & self._config.routes["run_arcane"]:
+                self._curr_loc = self._town_manager.refill_keys(self._curr_loc)
+                if not self._curr_loc:
+                	return self.trigger_or_stop("end_game", failed=True)
+            else: self._curr_loc = self._town_manager.repair_and_fill_tps(self._curr_loc)
```

Comment:
Should this be in an else branch? Couldnt we need to refill keys AND might have to repair?

Reply:
It refill key also repair and refill tps

Reply:
Thats my problem, cant we reuse the already implemented repair_and_tp to take a flag `refill_keys: bool = False`. It looks like pretty much duplicated code now in the town manager.

Reply:
since we are already at the vendor, why not repair and tp?